addon/adapters/rest.js

```diff
@@ -1009,6 +1101,253 @@ export default Adapter.extend(BuildURLMixin, {
+     * @param {Object} params
+     * @return {Object} request object
+     */
+    requestFor(params) {
+      const method = this.methodForRequest(params);
+      const url = this.urlForRequest(params);
+      const headers = this.headersForRequest(params);
+      const data = this.dataForRequest(params);
+
+      return { method, url, headers, data };
+    },
+
+    /**
+     * Convert a request object into a hash which can be passed to `jQuery.ajax`.
+     *
+     * @private
+     * @method requestToJQueryAjaxHash
+     * @param {Object} request
+     * @return {Object} jQuery ajax hash
+     */
+    requestToJQueryAjaxHash(request) {
+      const hash = {};
+
+      hash.type = request.method;
+      hash.url = request.url;
+      hash.dataType = 'json';
+      hash.context = this;
+
+      if (request.data) {
+        if (request.type !== 'GET') {
+          hash.contentType = 'application/json; charset=utf-8';
+          hash.data = JSON.stringify(request.data);
+        } else {
+          hash.data = request.data;
+        }
+      }
+
+      var headers = request.headers;
+      if (headers !== undefined) {
+        hash.beforeSend = function(xhr) {
+          Object.keys(headers).forEach((key) => xhr.setRequestHeader(key, headers[key]));
+        };
+      }
+
+      return hash;
+    },
+
+    /**
+     * Make a request using `jQuery.ajax`.
+     *
+     * @private
+     * @method makeRequest
+     * @param {Object} request
+     * @return {Promise} promise
+     */
+    makeRequest(request) {
+      const adapter = this;
+      const hash = this.requestToJQueryAjaxHash(request);
+
+      const { method, url } = request;
+      const requestData = { method, url };
+
+      return new Ember.RSVP.Promise((resolve, reject) => {
+
+        hash.success = function(payload, textStatus, jqXHR) {
+          let response = adapter.handleResponse(
+            jqXHR.status,
+            parseResponseHeaders(jqXHR.getAllResponseHeaders()),
+            payload,
+            requestData
+          );
+
+          if (response instanceof AdapterError) {
+            Ember.run.join(null, reject, response);
+          } else {
+            Ember.run.join(null, resolve, response);
+          }
+        };
+
+        hash.error = function(jqXHR, textStatus, errorThrown) {
+          let error;
+
+          if (errorThrown instanceof Error) {
+            error = errorThrown;
+          } else if (textStatus === 'timeout') {
+            error = new TimeoutError();
+          } else if (textStatus === 'abort') {
+            error = new AbortError();
+          } else {
+            error = adapter.handleResponse(
+              jqXHR.status,
+              parseResponseHeaders(jqXHR.getAllResponseHeaders()),
+              adapter.parseErrorResponse(jqXHR.responseText) || errorThrown,
+              requestData
+            );
+          }
+
+          Ember.run.join(null, reject, error);
+        };
+
+        Ember.$.ajax(hash);
```

Comment:
`this._ajaxRequest(hash);` to preserve a fastboot seam that was added after the original work for this pr was done.